kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -125,19 +134,48 @@ class ResidualBasedBlockBuilderAndSolver
         // Validate default parameters
         Parameters default_parameters = Parameters(R"(
         {
-            "name" : "ResidualBasedBlockBuilderAndSolver"
+            "name"                               : "ResidualBasedBlockBuilderAndSolver",
+            "block_builder"                      : true,
+            "diagonal_values_for_dirichlet_dofs" : "use_max_diagonal",
+            "silent_warnings"                    : false
         })" );
 
         ThisParameters.ValidateAndAssignDefaults(default_parameters);
+

```

Comment:
i would add:

~~~cpp
std::set<std::string> available_options_for_diagonal = {"no_scaling","use_max_diagonal","use_diagonal_norm","please write a number"};

if(available_options_for_diagonal.find(diagonal_values_for_dirichlet_dofs["smoother_type"].GetString()) == available_options_for_diagonal.end()) {
            msg << "Currently prescribed diagonal_values_for_dirichlet_dofs : " << ThisParameters["diagonal_values_for_dirichlet_dofs"].GetString() << std::endl;
            msg << "Admissible values are : no_scaling,use_max_diagonal,use_diagonal_norm,please write a number" << std::endl;
            KRATOS_ERROR << " available_options_for_diagonal is invalid: " << msg.str() << std::endl;
        }
~~~

Reply:
> i would add:
> 
> ```c++
> std::set<std::string> available_options_for_diagonal = {"no_scaling","use_max_diagonal","use_diagonal_norm","please write a number"};
> 
> if(available_options_for_diagonal.find(diagonal_values_for_dirichlet_dofs["smoother_type"].GetString()) == available_options_for_diagonal.end()) {
>             msg << "Currently prescribed diagonal_values_for_dirichlet_dofs : " << ThisParameters["diagonal_values_for_dirichlet_dofs"].GetString() << std::endl;
>             msg << "Admissible values are : no_scaling,use_max_diagonal,use_diagonal_norm,please write a number" << std::endl;
>             KRATOS_ERROR << " available_options_for_diagonal is invalid: " << msg.str() << std::endl;
>         }
> ```

this one

Reply:
Okay